rust/core-lib/src/editor.rs

```diff
@@ -762,6 +762,83 @@ impl<W: Write + Send + 'static> Editor<W> {
         self.insert(&*String::from(kill_ring_string));
     }
 
+    pub fn do_find(&mut self, chars: Option<&str>, case_sensitive: bool) {
+        use xi_rope::find::{find, CaseMatching};
+        
+        let search_string = chars.map(Cow::Borrowed).or_else(|| {
+            self.view.sel_regions().iter().last().and_then(|region| {
+                if region.is_caret() {
+                    None
+                } else {
+                    Some(Cow::Owned(self.text.slice_to_string(region.min(), region.max())))
+                }
+            })
+        });
+        
+        if search_string.is_none() {
+            return;
+        }
+
+        let search_string = search_string.unwrap();
+        if search_string.len() == 0 {
+            return
+        }
+
+        let case_matching = if case_sensitive {
+            CaseMatching::Exact
+        } else {
+            CaseMatching::CaseInsensitive
+        };
+
+        let mut sb = SpansBuilder::new(self.text.len());
+        let len = search_string.len();
+
+        let mut cursor = Cursor::new(&self.text, 0);
+        while let Some(pos) = find(&mut cursor, case_matching, &search_string) {
+            let style = Style { fg: 0xffc00000, font_style: 0 };
```

Comment:
so I guess for selections this is handled by the front-end. This should eventually be handled by some future `Theme` or `ColorScheme`. This seems okay, for now...